node/lib/util/synthetic_gc_util.js

```diff
@@ -37,13 +37,57 @@ const NodeGit = require("nodegit");
 const SubmoduleUtil = require("./submodule_util");
 const SubmoduleConfigUtil = require("./submodule_config_util");
 const SubmoduleFetcher = require("./submodule_fetcher");
+const SyntheticBranchUtil = require("./synthetic_branch_util");
 const UserError = require("./user_error");
 const GitUtil = require("./git_util");
+const fs = require("fs");
 
 const SYNTHETIC_BRANCH_BASE = "refs/commits/";
 
-let refExists = function(existingReferences, sha) {
-    return existingReferences.has(sha);
+let detail = {
+    /**
+     * Return whether sha is in existingReferences.
+     *
+     * @param {Set<String>} existingReferences
+     * @param {String}      sha
+     * @return {Boolean}
+     */
+    refExists : function(existingReferences, sha) {
+        return existingReferences.has(sha);
+    },
+
+    /**
+     * Remove gubbins from the specified 'path'.
+     *
+     * @param {String}      path
+     * @return {String}
+     */
+    rebuildUrl : function(path) {
+        path = path.replace(new RegExp("../main/"), "");
```

Comment:
This stuff needs to be configurable.  Also, dots in a regex are metacharacters; if you want literal dots, you need to escape them.